recap/src/Imp/Eval.hs

```diff
@@ -17,14 +20,37 @@ data Error = UndefinedVar String | ParsingErr String | DivByZero deriving (Show)
 -- Evaluation monad
 type EvalM = StateT VarMap (ExceptT Error IO)
 
+mapOp :: M.Map Op (Int -> Int -> Int)
+mapOp = M.fromList [(Plus, (+)), (Minus, (-)), (Mult, (*)), (Div, div)]
+
 evalExpr :: Expr -> EvalM Int
-evalExpr = undefined 
+evalExpr (Var x) = do
+  mapVar <- get
+  when (M.notMember x mapVar) $ lift (throwE $ UndefinedVar x)
+  return $ mapVar M.! x
+evalExpr (Const x) = return x
+evalExpr (BinOp op l r) = do
+  x <- evalExpr l
+  y <- evalExpr r
+  when (y == 0 && op == Div) $ lift (throwE DivByZero)
+  return $ (mapOp M.! op) x y
```

Comment:
It's better to implement a separate function for dealing with specific binary operators (separation of concerns).